provisioner/elevated.go

```diff
@@ -117,6 +118,14 @@ $result = $t.LastTaskResult
 if (Test-Path $log) {
     Remove-Item $log -Force -ErrorAction SilentlyContinue | Out-Null
 }
+
+$script = [System.Environment]::ExpandEnvironmentVariables("{{.ScriptFile}}")
+if (Test-Path $script) {
```

Comment:
Would be great to parameterize the cleanup with `SkipClean` but I don't see a clear way of doing this without breaking the interface or some hackery that might break the other provisioners. So going with the idea, if Packer created it and it has a password we should remove it :smile:

Reply:
Why not add "SkipClean" boolean to the elevatedOptions struct, and then use the golang templating to either add this func or not based on whether it't set? I don't see how that would break anything else if we default SkipClean to false. 

Something like 

```
{{ if not .SkipClean}}
$script = [System.Environment]::ExpandEnvironmentVariables("{{.ScriptFile}}")
if (Test-Path $script) {
    Remove-Item $script -Force -ErrorAction SilentlyContinue | Out-Null
}
$f = $s.GetFolder("\")
$f.DeleteTask("\$name", "")
{{ end }}
```

Reply:
oh, I see, nevermind. 

What a weird place for an interface.

I kind of doubt this interface is used outside of our Packer binary, so I wouldn't be too worried about breaking it. But I would probably wait for the 1.6 release. For now we can document that this cleanup is only skippable for non-elevated scripts.

Reply:
So that was the route I was going down, but then realized that the only entry path into the elevatedOptions from a provisioner is by calling the `GenerateElevatedRunner` func which takes an EleveatedProvisioner. I didn't want to update the other provisioners (puppet and chef) to add that field. But I guess I could check if the provisioner is a Powershell provisioner and take the value of SkipClean, leaving it false for other provisioners that implement the interface.